frontend/src/metabase/admin/people/components/SearchInput.jsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled, { css } from "styled-components";
+import { color } from "metabase/lib/colors";
+
+import Icon from "metabase/components/Icon";
+import { t } from "ttag";
+
+export default function SearchInput({
+  value,
+  placeholder,
+  onChange,
+  className,
+  clearButton,
+}) {
+  const handleClearClick = () => {
+    onChange("");
+  };
+
+  const showClearButton = clearButton && value.length > 0;
+
+  return (
+    <SearchFieldRoot className={className}>
+      <IconWrapper>
+        <Icon name="search" size={16} />
+      </IconWrapper>
+      <Input
+        clearButton
+        type="text"
+        placeholder={placeholder}
+        value={value}
+        onChange={e => onChange(e.target.value)}
+      />
+
+      {showClearButton && (
+        <ClearButton onClick={handleClearClick}>
+          <Icon name="close" size={12} />
+        </ClearButton>
+      )}
+    </SearchFieldRoot>
+  );
+}
+
+SearchInput.propTypes = {
+  onChange: PropTypes.func.isRequired,
+  placeholder: PropTypes.string,
+  value: PropTypes.string,
+  autoFocus: PropTypes.bool,
+  className: PropTypes.string,
+  clearButton: PropTypes.bool,
```

Comment:
`clearButton` prop makes me think it can also be a React node. Should we maybe prefix it like `hasClearButton`?

In general, I think it's a good practice to prefix boolean variables with verbs like "has" or "is"; it makes their type clear. Maybe it's worth adding to our style guide? Unless you agree